msteams-platform/messaging-extensions/build-api-based-message-extension.md

```diff
@@ -73,14 +113,206 @@ If the root object of the OpenAPI schema contains well-known array property name
 
 API-based message extensions are a potent tool that enhances your Teams app's functionality by integrating with external APIs. This enhances the capabilities of your app and provides a richer user experience. To implement message extension from an API, you need to follow these guidelines:
 
-* The `Commands.id` property in app manifest must match the corresponding `operationId` in the OpenAPI Description.
-* If a required parameter is without a default value, the command `parameters.name` in the app manifest must match the `parameters.name` in the OpenAPI Description.
+* The `Commands.id` property in app manifest must match the corresponding `operationId` in the OpenAPI Description. For example, if the operation ID is `getProduct`, the command ID must be `getProduct`.
```

Comment:
Since we are asking users to use Developer Portal or Toolkit to create an API-based ME and all these would be automatically configured. I've not covered the requirements under app manifest.

For Authentication, since there is no tool support until now, I've added the requirements in the app manifest update section for each auth types.